### PR TITLE
Rules table UX enhancements for 8.7

### LINKED ISSUE #2893: Rules table persistent state
## Description 

Related issue: https://github.com/elastic/kibana/issues/140263

The Rules table now retains certain display changes (such as filters, sorting, pagination) when the user navigates away and then returns (back button, navigate from a different page, etc.). Refer to related issue above for specific changes.

URLs are also persistent, so they can be shared and retain the same view when a new user follows the URL.

Opening a new tab or window does not retain the previous state.

## Docs to update

* [ ] [Manage detection rules | Sort and filter the rules list](https://www.elastic.co/guide/en/security/master/rules-ui-management.html#sort-filter-rules) - Update text, maybe also image above
* [ ] [Monitor and troubleshoot rule executions](https://www.elastic.co/guide/en/security/master/alerts-ui-monitor.html) - Update image, maybe also text

### LINKED ISSUE #2896: Updated toast notifications for skipped bulk edits to rules
## Description 

Related: https://github.com/elastic/kibana/issues/139802

Related PR above improves the toast message displayed when the user bulk edits multiple rules, but some of the rules are skipped. Uncertain if this affects docs, but we're creating this issue to determine that.

### LINKED ISSUE #2979: Rules table UX enhancements for 8.7
## Description 

Several enhancements related to filtering the Rules table (part of epic https://github.com/elastic/security-team/issues/1972):

* https://github.com/elastic/kibana/pull/150153
* https://github.com/elastic/kibana/pull/150059

Moar enhancements and screenshots to update (as reported in https://github.com/elastic/security-docs/issues/3005):

- https://github.com/elastic/kibana/issues/145968 ([PR](https://github.com/elastic/kibana/pull/150059))
- https://github.com/elastic/kibana/issues/132497 ([PR](https://github.com/elastic/kibana/pull/150153))
- https://github.com/elastic/kibana/issues/148196 ([PR](https://github.com/elastic/kibana/pull/149840))
- https://github.com/elastic/kibana/issues/138907 ([PR](https://github.com/elastic/kibana/pull/149840))

## Notes

- Check with team at feature sync for any other related enhancements.
- Will affect both screenshots and text instructions.

### PR DESCRIPTION
**Pull request by joepeeples**
Various enhancements to the Rules page and related functionality — bundled here since they all involve the same docs pages:
- Resolves #2893
- Resolves #2979
- Resolves #2896

Also: Confirmed that changes identified in #3005 have either been made in this PR or in other PRs.

Previews:
- [Manage detection rules | Sort and filter the rules list](https://security-docs_2998.docs-preview.app.elstc.co/guide/en/security/master/rules-ui-management.html#sort-filter-rules) - updated text and screenshot above this section
- [Manage detection rules | Modify existing rules settings](https://security-docs_2998.docs-preview.app.elstc.co/guide/en/security/master/rules-ui-management.html#edit-rules-settings) - expanded Note to explain skipped rules in bulk edits
- [Monitor and troubleshoot rule executions](https://security-docs_2998.docs-preview.app.elstc.co/guide/en/security/master/alerts-ui-monitor.html#rule-monitoring-tab) - updated screenshot only

### CODE REVIEW

**Comment by joepeeples** [comment]
Consider bundling #2979 in this PR, since it also has updates for the Rules table.

**Review by benironside — APPROVED** [review]
LGTM!

File: docs/detections/rules-ui-manage.asciidoc [hunk 1]
```diff
@@ -34,6 +34,10 @@ To filter the rules list, enter a search term in the search bar and press **Retu
 
 NOTE: Searches for index patterns and MITRE ATT&CK tactics and techniques must match exactly, are case sensitive, and do _not_ support wildcards. For example, to find rules using the `filebeat-*` index pattern, the search term `filebeat-*` is valid, but `filebeat` and `file*` are not because they don't exactly match the index pattern. Likewise, the MITRE ATT&CK tactic `Defense Evasion` is valid, but `Defense`, `defense evasion`, and `Defense*` are not.
 
+You can also filter the rules list by selecting the *Tags*, *Elastic rules*, *Custom rules*, *Enabled rules*, and *Disabled rules* filters next to the search bar.
```
Comment (nastasha-solomon) [inline docs/detections/rules-ui-manage.asciidoc:37]: @jmikell821 do you think the enabled and disabled UI text will be changed in 8.7? Or is this copy pretty much set for the release?

**Review by nastasha-solomon — APPROVED** [review]
I left one question for Janeen that might end up in some minor text and screenshot updates. Otherwise, everything LGTM!

**Review by jmikell821 — APPROVED** [review]
This all LGTM!

**Review by banderror — APPROVED** [review]
LGTM 👍
Thanks, Joe!

**Pull request merged**